### DISCUSSION TITLE
How am I supposed to update a split transaction using the API?

### DISCUSSION THREAD

**Discussion by jacopo-j** [firefly-iii/firefly-iii#5652] [Firefly III] [answered]
Let's say I have the following transaction stored on Firefly, which includes two split transactions (I removed irrelevant fields from the JSON for clarity):

```json
{
    "attributes":
    {
        "created_at": "2022-01-27T22:59:44+01:00",
        "group_title": "Some transaction group",
        "transactions":
        [
            {
                "amount": "7.750000000000000000000000",
                "date": "2022-01-27T00:00:00+01:00",
                "description": "First split",
                "destination_id": "311",
                "destination_name": "Destination account name",
                "destination_type": "Expense account",
                "order": 0,
                "source_id": "243",
                "source_name": "Source account name",
                "source_type": "Asset account",
                "tags": [],
                "transaction_journal_id": "1222",
                "type": "withdrawal",
            },
            {
                "amount": "7.740000000000000000000000",
                "date": "2022-01-27T00:00:00+01:00",
                "description": "Second split",
                "destination_id": "313",
                "destination_name": "Other destination account name",
                "destination_type": "Expense account",
                "order": 0,
                "source_id": "243",
                "source_name": "Source account name",
                "source_type": "Asset account",
                "tags": ["some tag"],
                "transaction_journal_id": "1333",
                "type": "withdrawal",
            }
        ],
        "updated_at": "2022-01-27T23:00:42+01:00",
    },
    "id": "1111",
    "type": "transactions"
}
```

Let's say I want to update the description of the first split transaction (ID = 1222) using the API, leaving the other split transaction as it is. How am I supposed to do that? I tried with two different methods so far, but neither of them works.

### Method 1

`PUT /api/v1/transactions/1111`

```json
{
    "apply_rules": false,
    "fire_webhooks": false,
    "transactions":
    [
        {
            "transaction_journal_id": 1222,
            "description": "Updated description"
        }
    ]
}
```

**Result**: the second split transaction (ID = 1333) vanishes and I'm left with only the (updated) first one.

### Method 2

`PUT /api/v1/transactions/1111`

```json
{
    "apply_rules": false,
    "fire_webhooks": false,
    "transactions":
    [
        {
            "description": "Updated description"
        },
        {}
    ]
}
```

**Result**: Internal Server Error, the whole transaction vanishes, already reported as issue #5610. Besides the bug, is this the correct method? If so, how am I supposed to know the "order" of the split transaction I want to edit if I only know its ID?

**Comment by JC5** [comment, marked as answer]
Sumbit just the `journal_id` for the splits you wish to leave unchanged. I'll expand the documentation for this.

**Reply by jacopo-j** [reply]
Ah, so the solution to my example would be:

```json
{
    "apply_rules": false,
    "fire_webhooks": false,
    "transactions":
    [
        {
            "transaction_journal_id": 1222,
            "description": "Updated description"
        },
        {
            "transaction_journal_id": 1333
        }
    ]
}
```

Is this correct? If so, it means that even if I need to update a single transaction split, I *must* know the IDs of all the other splits, right?

**Reply by JC5** [reply]
Yes. I'll double check this evening if this works as expected.

**Reply by jacopo-j** [reply]
Okay. Thanks for the clarification.

**Reply by JC5** [reply]
It won't work properly in 5.6.10, apologies about that. The next release should remedy that.

**Reply by jacopo-j** [reply]
For what it's worth, it worked correctly for me, although I only tried a couple times.

**Reply by JC5** [reply]
Nice, glad to hear it. I did find some issues myself so those at least will be fixed :)
